src/azure-cli/azure/cli/command_modules/appservice/_constants.py

```diff
@@ -39,39 +39,18 @@
         'java': '8'
     }
 }
-# functions version -> runtime -> runtime version : container image
-RUNTIME_TO_IMAGE_FUNCTIONAPP = {
+# functions version -> runtime : runtime versions
+RUNTIME_TO_SUPPORTED_VERSIONS_FUNCTIONAPP = {
     '2': {
-        'node': {
-            '8': 'mcr.microsoft.com/azure-functions/node:2.0-node8-appservice',
-            '10': 'mcr.microsoft.com/azure-functions/node:2.0-node10-appservice'
-        },
-        'python': {
-            '3.6': 'mcr.microsoft.com/azure-functions/python:2.0-python3.6-appservice',
-            '3.7': 'mcr.microsoft.com/azure-functions/python:2.0-python3.7-appservice'
-        },
-        'dotnet': {
-            '2': 'mcr.microsoft.com/azure-functions/dotnet:2.0-appservice'
-        },
-        'java': {
-            '8': 'mcr.microsoft.com/azure-functions/java:2.0-java8-appservice'
-        }
+        'node': ['8', '10'],
+        'python': ['3.6', '3.7'],
+        'dotnet': ['2'],
+        'java': ['8']
     },
     '3': {
-        'node': {
-            '10': 'mcr.microsoft.com/azure-functions/node:3.0-node10-appservice',
-            '12': 'mcr.microsoft.com/azure-functions/node:3.0-node12-appservice'
-        },
-        'python': {
-            '3.6': 'mcr.microsoft.com/azure-functions/python:3.0-python3.6-appservice',
-            '3.7': 'mcr.microsoft.com/azure-functions/python:3.0-python3.7-appservice',
-            '3.8': 'mcr.microsoft.com/azure-functions/python:3.0-python3.8-appservice'
-        },
-        'dotnet': {
-            '3': 'mcr.microsoft.com/azure-functions/dotnet:3.0-appservice'
-        },
-        'java': {
-            '8': 'mcr.microsoft.com/azure-functions/java:3.0-java8-appservice'
-        }
+        'node': ['10', '12'],
+        'python': ['3.6', '3.7', '3.8'],
+        'dotnet': ['3'],
```

Comment:
I do not think dotnet needs to be included here as Functions_Extension_Version App setting determines the which functions runtime version is used by the app. @ankitkumarr - can you confirm?

Reply:
That's a good point. My reasoning for leaving it here was that if a customer attempted to explicitly make a v2 function app with dotnet 3 (or something like that), they would get a clear error message as to why that fails.

Reply:
Per our discussion offline I've created this tracking issue: https://github.com/Azure/Azure-Functions/issues/1496 for these changes

Reply:
@gzuber, you are missing `PowerShell` version `6`. If this lookup table is only use for `Linux`, do you have a separate table for `Windows`?

Reply:
@pragnagopa, yes that's true that dotnet doesn't need to be here. @gzuber, yes throwing an error like that makes sense. I wonder if we should just throw an error if `--runtime-version` is set to any value for `--runtime dotnet`. Although that would be a breaking change and is not necessary.
So, I think this looks good, except maybe when setting linuxFxVersion for dotnet, I would just set `dotnet` to keep it consistent with other UX tools.